src/pybind/mgr/restful/common.py

```diff
@@ -96,11 +96,12 @@ def _gather_leaf_ids(node):
             return set([node['id']])
 
         result = set()
-        for item in node['items']:
+
+        for item in nodes_by_id[node['id']]['items']:
```

Comment:
nit, can we define a local variable, and use `reduce()` to collect its descendant nodes, something like
```py
node_id = node['id']
if node_id >= 0:
  return set(node_id)
children = nodes_by_id[node_id]['items']
return functools.reduce(operator.or_,
                        (_gather_leaf_ids(child) for child in children))
```

Reply:
This should work as well, it may harm the readability of the code though. AFAIK, both approaches should be pretty much on par in terms of complexity/speed.

You are right that we don't need the second if inside the for loop. Maybe I can remove that?

Reply:
yeah. that'd be better.

Reply:
OK, updated and repushed.

Reply:
oh, thought you agreed on defining a local variable for `node['id']`.